sale.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+"""
+    sale
+
+    Sale
+
+    :copyright: (c) 2015 by Openlabs Technologies & Consulting (P) Limited
+    :license: BSD, see LICENSE for more details.
+"""
+import dateutil.parser
+from decimal import Decimal
+
+from trytond.model import fields
+from trytond.transaction import Transaction
+from trytond.pool import PoolMeta, Pool
+
+
+__all__ = ['Sale']
+__metaclass__ = PoolMeta
+
+
+class Sale:
+    "Sale"
+    __name__ = 'sale.sale'
+
+    amazon_order_id = fields.Char(
+        'Amazon Order ID', readonly=True,
+        help="This is global and unique ID given to an order across whole"
+        "amazon"
+        "Warning: Editing this might result in duplicate orders on next"
+        " import"
+    )
+
+    @classmethod
+    def __setup__(cls):
+        """
+        Setup the class before adding to pool
+        """
+        super(Sale, cls).__setup__()
+
+        cls._sql_constraints += [
+            (
+                'amazon_id_unique',
+                'UNIQUE(amazon_order_id)',
+                'Sale with this amazon Order ID already exists'
+            )
+        ]
+
+    @classmethod
+    def find_or_create_using_amazon_id(cls, order_id):
+        """
+        This method tries to find the sale with the order ID
+        first and if not found it will fetch the info from amazon and
+        create a new sale with the data from amazon using
+        create_using_amazon_data
+
+        :param order_id: Order ID from amazon
+        :type order_id: string
+        :returns: Active record of sale order created/found
+        """
+        SaleChannel = Pool().get('sale.channel')
+
+        sales = cls.search([
+            ('amazon_order_id', '=', order_id),
+        ])
+        if sales:
+            return sales[0]
+
+        sale_channel = SaleChannel(
+            Transaction().context.get('sale_channel')
```

Comment:
`amazon_channel`

The current channel of the user could be anything. We want this order to go to the cooresponding amazon channel.